### ISSUE TITLE
Record with enum key type returns invalid OpenAPI

### ISSUE THREAD

**Issue by lnogol** [StefanTerdell/zod-to-json-schema#44]
```ts
z.record(z.enum(["a", "b", "c"]), z.number().int());
```

returns

```json
{
  "type": "object",
  "additionalProperties": {
    "type": "integer"
  },
  "propertyNames": {
    "enum": ["a", "b", "c"]
  }
}
```

for openapi target. 

This makes the result spec invalid. According to [this](https://swagger.io/docs/specification/data-models/keywords/), `propertyNames` is not supported in openapi

The same happens for any records actually, which return `"propertyNames": {}` leading to invalid spec

![image](https://user-images.githubusercontent.com/22595961/218574297-7509503a-a220-4140-88e5-4b530c069c47.png)

**Comment by lnogol** [comment]
Using a workaround for now:

```ts
export const objEnumKeys = <
	TEnum extends readonly [string, ...string[]],
	TSchema extends ZodSchema
>(
	values: TEnum,
	schema: TSchema
) => {
	return z
		.object(
			values.reduce((acc, val) => {
				acc[val as TEnum[number]] = schema;
				return acc;
			}, {} as Record<TEnum[number], TSchema>)
		)
		.partial();
};
```

```ts
objEnumKeys(["a", "b", "c"] as const, z.number().int());
```

returns

```json
{
  "type": "object",
  "properties": {
    "a": {
      "type": "integer"
    },
    "b": {
      "type": "integer"
    },
    "c": {
      "type": "integer"
    },
  },
  "additionalProperties": false
}
```

**Comment by StefanTerdell** [comment]
Fixed in 3.20.4
